server/src/main/java/io/druid/segment/indexing/granularity/UniformGranularitySpec.java

```diff
@@ -57,7 +57,9 @@ public UniformGranularitySpec(
     if (inputIntervals != null) {
       List<Interval> granularIntervals = Lists.newArrayList();
       for (Interval inputInterval : inputIntervals) {
-        Iterables.addAll(granularIntervals, this.segmentGranularity.getIterable(inputInterval));
```

Comment:
This is going to cause the created segments to be trimmed too. IMO we don't want that, since if someone says they want segmentGranularity DAY then they should get day-granular segments even if their intervals are not perfectly aligned to days.

So IMO the trimming should happen in GranularityPathSpec rather than here.